DotNet/Tenant/Services/ScheduleService.cs

```diff
@@ -46,103 +44,92 @@ public async Task StartAsync(CancellationToken cancellationToken)
+                        _logger.LogError("Facility {FacilityId} does not have a timezone set. Skipping scheduled jobs for this facility.", facility.FacilityId.SanitizeAndRemove());
                         continue;
                     }
-                    await AddJobsForFacility(facility, Scheduler);
+                    await AddJobsForFacility(facility, cancellationToken);
                 }
             }
 
             await Scheduler.Start(cancellationToken);
         }
 
-
         public async Task StopAsync(CancellationToken cancellationToken)
         {
             await Scheduler?.Shutdown(cancellationToken);
         }
 
-        public static async Task AddJobsForFacility(FacilityConfigModel facility, IScheduler scheduler)
+        public async Task AddJobsForFacility(Facility facility, CancellationToken cancellationToken = default)
         {
-            // create a job and trigger for monthly reports
+            var scheduler = await _schedulerFactory.GetScheduler(cancellationToken);
+
+            // Create a job and trigger for monthly reports
             if (facility.ScheduledReports.Monthly.Length > 0)
             {
-                createJobAndTrigger(facility, MONTHLY, scheduler);
+                await CreateJobAndTrigger(facility, MONTHLY, scheduler, cancellationToken);
             }
 
-            //create a job and trigger for weekly reports
+            // Create a job and trigger for weekly reports
             if (facility.ScheduledReports.Weekly.Length > 0)
             {
-                createJobAndTrigger(facility, WEEKLY, scheduler);
+                await CreateJobAndTrigger(facility, WEEKLY, scheduler, cancellationToken);
             }
 
-            // create a job and trigger for daily reports  
+            // Create a job and trigger for daily reports  
             if (facility.ScheduledReports.Daily.Length > 0)
             {
-                createJobAndTrigger(facility, DAILY, scheduler);
+                await CreateJobAndTrigger(facility, DAILY, scheduler, cancellationToken);
             }
-
         }
 
-        public static async Task DeleteJobsForFacility(String facilityId, IScheduler scheduler, List<string>? frequencies = null)
+        public async Task DeleteJobsForFacility(string facilityId, List<string>? frequencies = null, CancellationToken cancellationToken = default)
         {
-            frequencies ??= [ MONTHLY, WEEKLY, DAILY ];
-
-            if (frequencies != null)
-            {
-                foreach (String frequency in frequencies)
-                {
-                    var groupMatcher = GroupMatcher<JobKey>.GroupContains(nameof(KafkaTopic.ReportScheduled));
+            frequencies ??= new List<string> { MONTHLY, WEEKLY, DAILY };
 
-                    string jobKeyName = $"{facilityId}-{frequency}";
+            var scheduler = await _schedulerFactory.GetScheduler(cancellationToken);
```

Comment:
Can we use use the Scheduler property in all methods instead of calling GetScheduler  again.